src/BugsnagUnityForReview.mm

```diff
@@ -0,0 +1,876 @@
+    }     
+}
+
+const char * bugsnag_getSeverityFromEvent(const void *event){   
+    BSGSeverity theSeverity = ((__bridge BugsnagEvent *)event).severity;
+    if(theSeverity == BSGSeverityError)
+    {
+        return strdup("error");
+    }
+    if(theSeverity == BSGSeverityWarning)
+    {
+        return strdup("warning");
+    }
+    return strdup("info"); 
+}
+
+
+void bugsnag_getStackframesFromError(const void *error,const void *instance, void (*callback)(const void *instance, void *stackframes[], int stackframes_size)) {
+
+    NSArray * theStackframes = ((__bridge BugsnagError *)error).stacktrace;
+    void **c_array = (void **) malloc(sizeof(void *) * ((size_t)theStackframes.count));
+    for (NSUInteger i = 0; i < (NSUInteger)theStackframes.count; i++) {
+       c_array[i] = (__bridge void *)theStackframes[i];
+    }
+    callback(instance, c_array, [theStackframes count]);
+    free(c_array);
+}
+
+void bugsnag_getStackframesFromThread(const void *thread,const void *instance, void (*callback)(const void *instance, void *stackframes[], int stackframes_size)) {
+
+    NSArray * theStackframes = ((__bridge BugsnagThread *)thread).stacktrace;
+    void **c_array = (void **) malloc(sizeof(void *) * ((size_t)theStackframes.count));
+    for (NSUInteger i = 0; i < (NSUInteger)theStackframes.count; i++) {
+       c_array[i] = (__bridge void *)theStackframes[i];
+    }
+    callback(instance, c_array, [theStackframes count]);
+    free(c_array);
+}
+
+void bugsnag_getErrorsFromEvent(const void *event,const void *instance, void (*callback)(const void *instance, void *errors[], int errors_size)) {
+
+    NSArray * theErrors = ((__bridge BugsnagEvent *)event).errors;
+    void **c_array = (void **) malloc(sizeof(void *) * ((size_t)theErrors.count));
+    for (NSUInteger i = 0; i < (NSUInteger)theErrors.count; i++) {
+       c_array[i] = (__bridge void *)theErrors[i];
+    }
+    callback(instance, c_array, [theErrors count]);
+    free(c_array);
+}
+
+void bugsnag_getBreadcrumbsFromEvent(const void *event,const void *instance, void (*callback)(const void *instance, void *breadcrumbs[], int breadcrumbs_size)) {
+
+    NSArray * theBreadcrumbs = ((__bridge BugsnagEvent *)event).breadcrumbs;
+    void **c_array = (void **) malloc(sizeof(void *) * ((size_t)theBreadcrumbs.count));
+    for (NSUInteger i = 0; i < (NSUInteger)theBreadcrumbs.count; i++) {
+       c_array[i] = (__bridge void *)theBreadcrumbs[i];
+    }
+    callback(instance, c_array, [theBreadcrumbs count]);
+    free(c_array);
+}
+
+const char * bugsnag_getBreadcrumbMetadata(const void *breadcrumb) {
+    NSDictionary* sectionDictionary = ((__bridge BugsnagBreadcrumb *)breadcrumb).metadata;
+    return getMetadataJson(sectionDictionary);
+}
+
+void bugsnag_setBreadcrumbMetadata(const void *breadcrumb, const char *jsonString) {
+  if (jsonString == NULL)
+  {
+    return;
+  }
+  ((__bridge BugsnagBreadcrumb *)breadcrumb).metadata = getDictionaryFromMetadataJson(jsonString);
```

Comment:
Shouldn't this allow setting metadata to NULL?

```suggestion
  ((__bridge BugsnagBreadcrumb *)breadcrumb).metadata = jsonString  ? getDictionaryFromMetadataJson(jsonString) : nil;
```

Reply:
Done 👍